src/build/build.ts

```diff
@@ -12,48 +12,38 @@
  * http://polymer.github.io/PATENTS.txt
  */
 
-import * as del from 'del';
 import * as path from 'path';
 import * as logging from 'plylog';
 import {dest} from 'vinyl-fs';
 import mergeStream = require('merge-stream');
 import {PolymerProject, addServiceWorker, SWConfig} from 'polymer-build';
 
 import {OptimizeOptions, getOptimizeStreams} from './optimize-streams';
-import {ProjectConfig} from 'polymer-project-config';
+import {ProjectConfig, ProjectBuildOptions} from 'polymer-project-config';
 import {PrefetchTransform} from './prefetch';
 import {waitFor, pipeStreams} from './streams';
 import {loadServiceWorkerConfig} from './load-config';
 
 const logger = logging.getLogger('cli.build.build');
 
-const buildDirectory = 'build/';
-
-export interface BuildOptions extends OptimizeOptions {
-  addServiceWorker?: boolean;
-  swPrecacheConfig?: string;
-  insertPrefetchLinks?: boolean;
-  bundle?: boolean;
-};
-
 export async function build(
-    options: BuildOptions, config: ProjectConfig): Promise<void> {
+    options: ProjectBuildOptions, config: ProjectConfig): Promise<void> {
```

Comment:
Agreed, will add comments. 

Remember that the goal is for `build()` to handle all builds defined in the `config`, and not take a single `BuildOptions` at all. (builds defined in CLI flags may complicate this a bit). That way we'd only need a single `PolymerProject` instance. But this is currently blocked by the split/rejoin bug. 

Will summarize all this in the added comments.